[fix] cicid 트러블 슈팅

## 🤔 오류 내용

Run appleboy/ssh-action@master
Run echo "$GITHUB_ACTION_PATH" >> $GITHUB_PATH
Run entrypoint.sh
Downloading drone-ssh-1.8.2-linux-amd64 from https://github.com/appleboy/drone-ssh/releases/download/v1.8.2
======= CLI Version Information =======
Drone SSH version 1.8.2
=======================================
docker: 'docker pull' requires 1 argument
Usage:  docker pull [OPTIONS] NAME[:TAG|@DIGEST]
Run 'docker pull --help' for more information
2025/12/22 12:52:10 Process exited with status 1
Error: Process completed with exit code 1. 
<br>

## ⚠ 에러 캡쳐

<br>

--- a/.github/workflows/CICD.yml
+++ b/.github/workflows/CICD.yml
@@ -4,10 +4,10 @@ name: EveryBuddy server CI & CD
 on:
   push:
     branches:
-      - main
+      - dev
   pull_request:
     branches:
-      - main
+      - dev
     
 
 

--- a/.github/workflows/CICD.yml
+++ b/.github/workflows/CICD.yml
@@ -59,3 +59,30 @@ jobs:
           cache-from: type=gha
           cache-to: type=gha,mode=max
           
+  deploy: 
+    needs: build_and_push
+    if: github.event_name == 'push' && github.ref == 'refs/heads/develop'  
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Deploy to EC2
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.EC2_HOST }}
+          username: ${{ secrets.EC2_USERNAME }}
+          key: ${{ secrets.EC2_SSH_KEY }}
+          script: |
+            docker pull ${{ secrets.DOCKER_USERNAME }}/everybuddy:latest
+            docker stop everybuddy-backend 2>/dev/null || true
+            docker rm everybuddy-backend 2>/dev/null || true
+            docker run -d \
+              --name everybuddy-backend \
+              -p 8080:8080 \
+              --restart unless-stopped \
+              ${{ secrets.DOCKER_USERNAME }}/everybuddy:latest
+            docker image prune -f
+
+      - name: Health Check
+        run: |
+          sleep 20
+          curl -f http://${{ secrets.EC2_HOST }}:8080/actuator/health        

